src/sys/wait.rs

```diff
@@ -112,149 +96,52 @@ impl WaitStatus {
     }
 }
 
-#[cfg(any(target_os = "linux",
-          target_os = "android"))]
 mod status {
     use sys::signal::Signal;
-    use libc::c_int;
-    use libc::SIGTRAP;
+    use libc::{self, c_int, SIGTRAP};
 
     pub fn exited(status: i32) -> bool {
-        (status & 0x7F) == 0
+        unsafe { libc::WIFEXITED(status) }
```

Comment:
Likely just an oversight.  We could move them out of the `f!` block in libc and then they wouldn't be `unsafe` anymore.  However, I wonder if they even belong in libc at all.  There aren't many functions in libc that have more than a simple FFI call.